src/OutputManager.C

```diff
@@ -0,0 +1,160 @@
+// Copyright 2017 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS), National Renewable Energy Laboratory, University of Texas Austin,
+// Northwest Research Associates. Under the terms of Contract DE-NA0003525
+// with NTESS, the U.S. Government retains certain rights in this software.
+//
+// This software is released under the BSD 3-clause license. See LICENSE file
+// for more details.
+//
+
+#include <OutputManager.h>
+#include <NaluParsing.h>
+#include <stk_io/StkMeshIoBroker.hpp>
+#include <stk_mesh/base/MetaData.hpp>
+#include <stk_util/environment/WallTime.hpp>
+#include <stk_util/parallel/ParallelReduce.hpp>
+#include <string>
+#include <set>
+
+namespace sierra {
+namespace nalu {
+void
+OutputManager::load(const YAML::Node& node)
+{
+  std::vector<const YAML::Node*> foundNodes;
+  NaluParsingHelper::find_nodes_given_key("output", node, foundNodes);
+
+  const int nFound = foundNodes.size();
+  infoVec_.resize(nFound);
+  indexVec_.resize(nFound);
+
+  for (int i = 0; i < nFound; i++) {
+    infoVec_[i].load(*foundNodes[i]);
+  }
+}
+
+void
+OutputManager::create_output_mesh(
+  stk::io::StkMeshIoBroker* ioBroker, stk::mesh::MetaData* metaData)
```

Comment:
I'm assuming that the meta data passed in is associated with the bulk data on the ioBroker. If that is the case, you can grab the bulk data from the ioBroker and then grab the meta data from the bulk data

Reply:
ah nice! thanks @tokusanya